packages/ember-glimmer/tests/integration/helpers/unbound-test.js

```diff
@@ -0,0 +1,562 @@
+
+    this.render(`<ul>{{#each view.items as |item|}}<li>{{unbound item}}</li>{{/each}}</ul>`, {
+      items: emberA(['a', 'b', 'c', 1, 2, 3])
+    });
+
+    this.assertText('abc123');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('abc123');
+
+    this.runTask(() => set(this.context.items, 'firstObject', 'hello'));
+
+    this.assertText('abc123');
+
+    this.runTask(() => set(this.context, 'items', emberA(['a', 'b', 'c', 1, 2, 3])));
+
+    this.assertText('abc123');
+  }
+
+  ['@htmlbars should be able to use unbound helper in #each helper (with objects)']() {
+
+    this.render(`<ul>{{#each view.items as |item|}}<li>{{unbound item.wham}}</li>{{/each}}</ul>`, {
+      items: emberA([{ wham: 'bam' }, { wham: 1 }])
+    });
+
+    this.assertText('bam1');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('bam1');
+
+    this.runTask(() => this.context.items.setEach('wham', 'HEY'));
+
+    this.assertText('bam1');
+
+    this.runTask(() => set(this.context, 'items',  emberA([{ wham: 'bam' }, { wham: 1 }])));
+
+    this.assertText('bam1');
+  }
+
+  ['@htmlbars it should assert unbound cannot be called with multiple arguments']({ throws }) {
+
+    let willThrow = () => {
+      this.render(`{{unbound foo bar}}`, {
+        foo: 'BORK',
+        bar: 'BLOOP'
+      });
+    };
+
+    throws(willThrow, /unbound helper cannot be called with multiple params or hash params/);
+  }
+
+  ['@htmlbars should render on attributes']() {
+
+    this.render(`<a href="{{unbound foo}}"></a>`, {
+      foo: 'BORK'
+    });
+
+    this.assertHTML('<a href="BORK"></a>');
+
+    this.runTask(() => this.rerender());
+
+    this.assertHTML('<a href="BORK"></a>');
+
+    this.runTask(() => set(this.context, 'foo', 'OOF'));
+
+    this.assertHTML('<a href="BORK"></a>');
+
+    this.runTask(() => set(this.context, 'foo', 'BORK'));
+
+    this.assertHTML('<a href="BORK"></a>');
+  }
+
+  ['@htmlbars syntax form updates on parent re-render']() {
+
+    let fooBarInstance;
+    let FooBarComponent = Component.extend({
+      init() {
+        this._super(...arguments);
+        fooBarInstance = this;
+      },
+      foo: 'BORK',
+      barBinding: 'foo'
+    });
+
+    this.registerComponent('foo-bar', {
+      ComponentClass: FooBarComponent,
+      template: `{{unbound foo}} {{unbound bar}}`
+    });
+
+    this.render(`{{foo-bar}}`);
+
+    this.assertText('BORK BORK');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('BORK BORK');
+
+    this.runTask(() => {
+      set(fooBarInstance, 'foo', 'OOF');
+      this.rerender();
+    });
+
+    this.assertText('OOF OOF');
+
+    this.runTask(() => {
+      set(fooBarInstance, 'foo', 'BLIP');
+    });
+
+    this.assertText('OOF OOF');
+
+    this.runTask(() => {
+      set(fooBarInstance, 'foo', 'BORK');
+      this.rerender();
+    });
+
+    this.assertText('BORK BORK');
+  }
+
+  ['@htmlbars sexpr form does not update on parent re-render']() {
```

Comment:
Is it just me or is it weird that sexprs and syntax act differently when the parent re-renders?